changelogs/fragments/57221-nios-host-record-bug-fix.yaml

```diff
@@ -0,0 +1,3 @@
+bugfixes:
+- This PR fixes the issue raised where idempotency was failing when DNS bypassing was set to False and also exception
+  error faced in nios_host_reord - https://github.com/ansible/ansible/pull/57221.
```

Comment:
Note, You don't need to say "This PR" in a changelog.  The end user will read the changelog as a part of the entire release rather than associated with a specific PR.  I believe that common style would be something like
> Fix a traceback and idempotency in nios_host_record when DNS bypassing was set to False. https://github.com/ansible/ansible/pull/57221

Not a blocker for the backport

Reply:
@abadger Thanks and duly noted